packages/genesys-spark-components/src/components/stable/gux-listbox/option-group/gux-option-group.tsx

```diff
@@ -9,17 +12,103 @@ import { Component, h, Host, JSX, Prop } from '@stencil/core';
   shadow: true
 })
 export class GuxOptionGroup {
+  private rootParent: HTMLElement;
+  private parentObserver: MutationObserver;
+
+  @Element() root: HTMLGuxOptionGroupBetaElement;
+
   @Prop()
   label!: string;
 
   // @Prop()
   // disabled: boolean;
   // TOOD: Add disabled state once design work is complete
 
+  @State()
+  private parentFilterValue: string = '';
+
+  @State()
+  filtered: boolean = false;
+
+  @State()
+  hideDivider: boolean = false;
+
+  componentDidLoad(): void {
+    this.rootParent = getClosestElement(
+      'gux-dropdown',
+      this.root
+    ) as HTMLElement;
+
+    this.parentObserver = onMutation(this.rootParent, () => {
+      this.hideDivider =
+        this.parentFilterValue.length > 0 &&
```

Comment:
I think you want a function here called something like `hasVisibleNextSibling`. If true you show the divider if false you hide the divider.